pkg/controllers/managementuser/alert/deployer/upgradeimpl.go

```diff
@@ -397,12 +397,13 @@ func (l *AlertService) removeFinalizerFromLegacyAlerting() error {
 	}
 
 	for _, v := range oldProjectAlert {
+		if len(v.Finalizers) == 0 {
+			continue
+		}
 		newObj := v.DeepCopy()
 		newObj.SetFinalizers([]string{})
-		if !reflect.DeepEqual(newObj, v) {
```

Comment:
Why is has the DeepEqual been removed?

Reply:
Because the len(v.Finalizers) checks the only thing we are modifying.  I believe that check always succeeds because `nil` != `[]string{}`